### DISCUSSION TITLE
Trasferire Homeassistant-OS su minipc e docker win11

### DISCUSSION THREAD

**Discussion by calz64** [indomus/forum#2075] [HUB: Home Assistant]
Buongiorno, ho iniziato HA un paio di anni fa con una istallazione su raspy4 > HA-OS. Il pacchetto funziona benone e ho automatizzato un sacco di cose, tanto che non posso perdere/rifare tutto. Vorrei portare tutto su un minipc Herobox e attivare docker per avere altri servizi (tipo nextcloud). 
Il problema è che ho intallato HA su docker win11 seguendo la guida (ovviamente si tratta di HA core), ma ora come faccio a portare tutto il lavoro della HA-OS su minipc.

**Comment by snakuzzo** [comment]
Puoi riportare tutto quello che hai fatto su Core...semplicemente recuperando tutto quello che hai nella directory `/config` (dovrai sicuramente sistemare qualche cosina qua e là a mano ... tipo eventuali percorsi, etc).
Per quanto riguarda gli addon...non esistono su Core. Quindi dovrai crearti i vari servizi a parte, magari come container docker separati.
Nel caso ti consiglio l'utilizzo del plugin `compose` di docker o `docker-compose`.
Ad ogni modo, io eviterei di usare Windows...ma mi butterei su linux.
Se proprio vuoi usare HA OS su Windows, scarica l'immagine della VM che trovi sul sito ufficiale e la utilizzi con VirtualBox.

**Reply by calz64** [reply]
Ho provato debian11, premetto sono un estimatore di linux però trovo desktop-docker di win un sollievo, vedere i processi sarà ingenuo ma mi da sollievo. In debian codice a rullo e poi errore apache!!! non riesco a risolvere, azz.. Anche compose non ho tanto capito: le cartelle docker-compose alla fine non le ho trovate, forse non ho capito granchè: ovviamente per linux non esiste un docker-desktop?

**Reply by snakuzzo** [reply]
No comment

**Reply by calz64** [reply]
capisco non commentare la frustazione... ;) ma per linux un piccolo aiutino, una GUI-docker?

**Reply by snakuzzo** [reply]
> prova direttamente haos in una macchina virtuale, se il minipc ha abbastanza potenza andrà meglio di un raspberry...

Non è quello che ha chiesto

**Reply by snakuzzo** [reply]
> capisco non commentare la frustazione... ;) ma per linux un piccolo aiutino, una GUI-docker?

Hai cercato su Google? 

Primo risultato...

https://www.docker.com/blog/the-magic-of-docker-desktop-is-now-available-on-linux/

**Reply by calz64** [reply]
giuro ho provato, nulla: devi avere un motore "tuning" (codice modificato da te) <.)

**Reply by snakuzzo** [reply]
Dovresti leggere la documantazione ufficiale [QUI](https://docs.docker.com/desktop/install/linux-install/)
Ad ogni modo, cerca su google "docker desktop alternatives linux" ... e dai una letta qua e là. Es. [uno dei risultati](https://medium.com/@oribenhur/a-better-alternative-for-docker-desktop-3e8fa38d618)

E comunque...Docker Desktop non serve veramente a niente.

**Comment by calz64** [comment]
Dopo varie prove home assistant supevised è installato su docker (Debian 11 e mini pc-Herobox) come da guida indomus. Ho fatto upload del backup della HA_OS che avevo su raspy e incredibilmente c'è tutto, funziona più o meno tutto, devo ancora passare da raspy e provare le antenne zigbee e zwave (cioè non vedo alcuni sensori) ma tutto ciò che e wifi è OK. 
Però ci sono problemi con l'host io credo, mi da queste riparazioni (vedi allegato) che ho cercato di capire, mi sembrano legate a problemi con l'Host. 
Secondo voi si riparano, o si deve reinstallare, qui dice che si risolve con script e reinstallando: https://github.com/home-assistant/supervised-installer

Grazie un saluto.
![Schermata del 2022-11-18 17-10-55](https://user-images.githubusercontent.com/84615011/202751838-23b90caa-b4e3-4cd0-8085-74ecf2933465.png)

**Reply by snakuzzo** [reply]
In teoria, avendo usato Debian 11, non dovresti avere problemi legati ad OS non supportati.
Dalle note ufficiali che puoi trovare [QUI](https://github.com/home-assistant/supervised-installer), cito...

> _We only accept bug reports for issues that have been reproduced on a freshly installed, fully updated Debian with no additional packages._


Non è che per caso hai installato qualcosa a parte che possa dar fastidio ?

Dovresti provare a controllare i log del supervisor

**Reply by calz64** [reply]
Cosa può succedere se si continua ad avere queste riparazioni 'aperte'?

**Reply by snakuzzo** [reply]
Che esplode tutto? :)
Scherzo.
Hai toppato in qualche punto.

Anche un altro utente in chat aveva problemi (derivati da altre esigenze). La nostra guida è un po' datata e va aggiornata (lo faremo a giorni).
Per ora quello che ti consiglio di fare è ripassare gli step seguendo quella ufficiale che puoi trovare [QUI](https://github.com/home-assistant/supervised-installer)

**Reply by calz64** [reply]
Ok grazie, metto in campo gli sminatori..;) 
Essendo alla mia prima esperienza docker vorrei chiederti : 
1 - mi sembra di capire che anche se ho HA installato, i comandi che la guida indica di fare, ripuliranno e rifaranno il tutto, giusto?
2 - cosa significa dover mantener aggiornati tutti i componenti richiesti, bisogna fare a mano quello che HA_OS fa in automatico? ci sono guide in questo senso?
Grazie.

**Reply by snakuzzo** [reply]
1. Il .deb non l'ho creato io...prova
2. Di che parli?

**Reply by calz64** [reply]
punto - 2. di questo passaggio della guida.
`By not using the Home Assistant Operating System, the user is responsible for making sure that all required components are installed and maintained. Required components and their versions will change over time. Home Assistant Supervised is provided as-is as a foundation for community supported do-it-yourself solutions. We only accept bug reports for issues that have been reproduced on a freshly installed, fully updated Debian with no additional packages.`

**Reply by snakuzzo** [reply]
Ti dice in maniera generica che, nonostante ci siano dei prerequisiti, con il tempo tutto potrebbe cambiare (OS incluso) ed è affidato all'utente il compito di tenersi al passo... Non come su HA OS dove clicchi e ti aggiorna tutto in automatico, anche l'OS.
Esempio...
Se domani mattina esce la nuova stable di Debian, nel giro di qualche mese (4 se non ricordo male) devi aggiornare OS ed eventualmente tutti i pacchetti necessari.
Se rimani su versioni vecchie e per caso segnali un problema legato a qualcosa che non hai aggiornato, nessuno ufficialmente ti darà supporto...te la suoni e te la canti da solo.
Attento anche a quello che installi oltre a quello indicato in guida...se crea conflitti con HA Supervised, sarà solo un problema tuo.

**Reply by calz64** [reply]
ok ho capito perfettamente. grazie ancora

**Reply by calz64** [reply]
Buongiorno, è andato tutto a posto, ho seguito i passaggi, ma scusa se approfitto della tua pazienza per capire gdi più Docker:

**Step1 - Ho installato delle dipendenze sull'Host, giusto?**
```
apt-get install \
apparmor \
jq \
wget \
curl \
udisks2 \
libglib2.0-bin \
network-manager \
dbus \
systemd-journal-remote -y
```
** poi ho seguito Step 2 e 3:  -  Installo Docker-CE e OS-Agent:**

#Infine Step 4: - qui Installo la nuova immagine di HA-supervised,  ma com'è che accade che mi trovo il container'' di docker gia tutto configurato e sistemato:
```
wget https://github.com/home-assistant/supervised-installer/releases/latest/download/homeassistant-supervised.deb
dpkg -i homeassistant-supervised.deb
```

**Reply by snakuzzo** [reply]
Guarda che il pacchetto non è che "installa l'immagine di HA Supervised". Non esiste alcuna immagine Docker di HA Supervised :).
E' il tipo di installazione che è chiamata "HA Supervised su Docker" e consiste nello scaricare, tramite quel pacchetto .deb, le immagini Docker necessarie a far partire tutto il cucuzzaro, istanziare i container, etc.

**Comment by KC-inDomus** [comment]
> Buongiorno, ho iniziato HA un paio di anni fa con una istallazione su raspy4 > HA-OS. Il pacchetto funziona benone e ho automatizzato un sacco di cose, tanto che non posso perdere/rifare tutto. Vorrei portare tutto su un minipc Herobox e attivare docker per avere altri servizi (tipo nextcloud). Il problema è che ho intallato HA su docker win11 seguendo la guida (ovviamente si tratta di HA core), ma ora come faccio a portare tutto il lavoro della HA-OS su minipc.

domanda cretina perché non ho mai provato, ma fare un backup di core da HassOS (non il backup di HassOS, ma solo quello di Core) e ripristinarlo su Core@Docker?

**Reply by snakuzzo** [reply]
Inizialmente gli avevo consigliato di usare Core su Docker, ma poi lui ha scelto HA Supervised per il discorso degli addon.
Comunque sì...il backup su HA OS permette di fare il backup parziale, quindi anche solo di Core, che alla fine equivale ad impacchettare la dir `/config`

![image](https://user-images.githubusercontent.com/7205371/203728117-22d6a71d-17a7-424b-9c8a-e06595be889c.png)

**Comment by calz64** [comment]
Grazie, in effetti è stato un inferno durato un paio di settimane, ma grazie anche al vostro (snakuzzo-docet) aiuto, ho finito la preparazione della macchina minipc-debian11, con HA_supervised_Docker (che è un pacchetto .deb molto comodo per me che ancora non padroneggio Docker, anzi non ho capito proprio una mazza! speriamo in futuro di migliorare), e nexcloud-mit-nginx grazie ad una guida in tedesco (translate.google-docet), guida ben fatta a prova di....
Si imparano molte cose sperimentando, o forse è solo un'illusione, comunque funziona, per ora! grazie ancora.
